lib/src/main/java/com/knuddels/jtokkit/TokenEncoderLarge.java

```diff
@@ -0,0 +1,109 @@
+package com.knuddels.jtokkit;
+
+
+import java.util.List;
+import java.util.TreeMap;
+
+import static com.knuddels.jtokkit.TokenEncoder.MAX_RANK;
+
+final class TokenEncoderLarge {
```

Comment:
Alternative byte pair merge with basically linear-time merge for DOS-type worst-case-scenarios.